mobile/src/test/java/org/openhab/habdroid/model/OpenHAB2WidgetTest.java

```diff
@@ -1,67 +1,404 @@
+    @Before
+    public void testCreateOpenHABWidgetFromJson_createsOpenHAB2Widget() throws Exception {
+        sut1 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(1), "PNG");
+        sut2 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(2), "SVG");
+        sut3 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(3), "SVG");
+        assertTrue(sut1 instanceof OpenHAB2Widget);
+        assertTrue(sut2 instanceof OpenHAB2Widget);
     }
 
     @Test
-    public void testCreateOpenHABWidgetFromJson_createsOpenHAB2Widget() throws Exception {
-        OpenHABWidget sut = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(), "PNG");
-        assertTrue(sut instanceof OpenHAB2Widget);
-    }
-
-    private JSONObject createJSONObject() throws Exception {
-
-        String json = "{\n" +
-                "    \"widgetId\": \"demo\",\n" +
-                "    \"type\": \"Group\",\n" +
-                "    \"label\": \"Group1\",\n" +
-                "    \"icon\": \"groupicon\",\n" +
-                "    \"url\": \"http://localhost/url\",\n" +
-                "    \"minValue\": \"0.0\",\n" +
-                "    \"maxValue\": \"10.0\",\n" +
-                "    \"step\": \"1\",\n" +
-                "    \"refresh\": \"10\",\n" +
-                "    \"period\": \"D\",\n" +
-                "    \"service\": \"D\",\n" +
-                "    \"height\": \"10\",\n" +
-                "    \"iconcolor\": \"white\",\n" +
-                "    \"labelcolor\": \"white\",\n" +
-                "    \"valuecolor\": \"white\",\n" +
-                "    \"mappings\": [{\n" +
-                "      \"command\": \"ON\",\n" +
-                "      \"label\": \"On\"\n" +
-                "    }],\n" +
-                "    \"item\": {\n" +
-                "      \"type\": \"GroupItem\",\n" +
-                "      \"name\": \"group1\",\n" +
-                "      \"state\": \"OFF\",\n" +
-                "      \"link\": \"http://localhost/rest/items/group1\"\n" +
-                "    },\n" +
-                "    \"linkedPage\": {\n" +
-                "      \"id\": \"0001\",\n" +
-                "      \"title\": \"LinkedPage\",\n" +
-                "      \"icon\": \"linkedpageicon\",\n" +
-                "      \"link\": \"http://localhost/rest/sitemaps/demo/0001\",\n" +
-                "      \"leaf\": \"false\"\n" +
-                "    },\n" +
-                "    \"widgets\": [{ \"widgetId\": \"demo11\" }]\n" +
-                "  }";
+    public void testGetIconPath() throws Exception {
+        assertEquals("icon/groupicon?state=OFF&format=PNG", sut1.getIconPath());
+        assertEquals("icon/groupicon?state=ON&format=SVG", sut2.getIconPath());
+        assertEquals("icon/slider?state=81&format=SVG", sut3.getChildren().get(0).getIconPath());
+
+        // Rollersutter icon should always be 0 to 100, not ON/OFF
```

Comment:
If you've a comment for the assert, I would put this comment into the message, so when this assert fails, you get this message and probably know better why the test failed or what was expected instead :)

Reply:
How can I do that?

Reply:
For all assert* methods provided by JUnit (and most of ones provided by other frameworks) overload the assert* method to take a string as a first parameter, the expected result as the second and the current result as the third parameter (or only the current result, if the method itself contains the expectation, like assertTrue). That means, you can write:

````
assertEquals("Rollersutter icon should always be 0 to 100, not ON/OFF", "icon/slider?state=81&format=SVG", sut3.getChildren().get(0).getIconPath());
````
And, if the test fails, the output would be something like:
````
junit.framework.ComparisonFailure: Rollersutter icon should always be 0 to 100, not ON/OFF 
Expected :icon/slider?state=81&format=SVG
Actual   :
 <Click to see difference>


	at junit.framework.Assert.assertEquals(Assert.java:100)
	at org.openhab.habdroid.core.connection.ConnectionFactoryTest.testGetAnyConnectionWifiLocalRemote(ConnectionFactoryTest.java:157)
// ...

Process finished with exit code -1
````
So, the comment is still there in the code and also visible in the test output :)

Reply:
Ok, I changed it. I was anyway looking for a way how to easily see which test failed (in travis)